src/content/docs/en/reference/api-reference.mdx

```diff
@@ -2075,6 +2149,118 @@ export const onRequest = defineMiddleware(async (context, next) => {
 })
 ```
 
+## Actions (astro:actions)
+
+<p>
+<Since v="4.15.0" />
+</p>
+
+Actions are backend functions used for type-safe server-to-client communication. All utilities to define and call actions are exposed by the `astro:actions` module. For examples and usage instructions, [see the Actions guide](/en/guides/actions/).
+
+### `defineAction()`
+
+<p>
+<Since v="4.15.0" />
+</p>
+
+The `defineAction()` utility is used to define new actions from the `src/actions/index.ts` file. This accepts a `handler()` function containing the server logic to run, and an optional `input` property to validate input parameters at runtime.
+
+```ts
+export const server = {
+    getGreeting: defineAction({
+    input: z.object({
+      name: z.string(),
+    }),
+    handler: async (input) => {
+      return `Hello, ${input.name}!`
+    }
+  })
+}
+```
+
+#### `handler()` property
+
+<p>
+<Since v="4.15.0" />
+</p>
+
+`defineAction()` accepts a `handler()` function containing the server logic to run when the action is called. This function can return data that is automatically serialized and sent to the caller.
+
+Return values are parsed using the [devalue library](https://github.com/Rich-Harris/devalue). This supports JSON values, along with instances of `Date()`, `Map()`, `Set()`, or `URL()`.
+
+#### `input` validator
+
+<p>
+<Since v="4.15.0" />
+</p>
+
+The optional `input` property accepts a Zod validator to validate handler inputs at runtime. If the action fails to validate, [a `BAD_REQUEST` error](#actionerror) is returned and the `handler` is not called.
+
+If used with `accept: 'form'`, `input` must use the `z.object()` validator.
+
+Extension functions including `.refine()`, `.transform()`, and `.pipe()` are also supported on this object. The following validators are supported for form data fields:
+
+- Inputs of type `number` can be validated using `z.number()`
+- Inputs of type `checkbox` can be validated using `z.boolean()`
+- Inputs of type `file` can be validated using `z.instanceof(File)`
+- Multiple inputs of the same `name` can be validated using `z.array(/* validator */)`
+- All other inputs can be validated using `z.string()`
+
+### `isInputError()`
+
+<p>
+
+**Type:** `<T>(error?: unknown | ActionError<T>) => error is ActionInputError<T>`<br/>
```

Comment:
After reading Sarah's suggestion (see [this discussion](https://github.com/withastro/docs/pull/8980#discussion_r1733200551)), I think that the type I proposed may not be clear enough for those who are not familiar with Typescript.

Also I didn't realized that we have a section about `ActionError`.

So maybe we should just rewrite it this way:

```suggestion
**Type:** <code>(error?: unknown | <a href="#actionerror">ActionError</a>) => boolean</code><br/>
```

In summary: we should removed Typescript syntax and add a link to the ActionError heading. Those familiar with Typescript will be happy to find that it is a type-guard when using it, others don't need to wonder what the return type is.

What do you think?